### ISSUE TITLE
Unsubscribe ApolloObservable subscriptions on ngOnDestroy

### ISSUE THREAD

**Issue by manuelfink** [apollostack/angular2-apollo#216]
I think angular2-apollo is not automatically unsubscribing from queries on ngOnDestroy (React componentWillUnmount), there as react-apollo is doing this. I guess unsubscribing is an important feature to avoid fetching data which is not needed any more.

I run into problems with this, implementing an app logout. I tried to reset the apolloStore to avoid that one users should not see data from previous user.  (Right now I work around by reloading the page.) @helfer came up with the suggestion to use `apolloclient.resetStore` which currently is not exposed through angular2-apollo (which I expose through my service providing apollo). However resetStore refetches all previously fired queries. This results in the problems and errors and an hangup of ng2 since the user is unauthorized and therefore gets an 401 unauthorized message.

Or is there any other best practice to clear the store?

Discussion on apollo-client https://github.com/apollostack/apollo-client/issues/1075

**Comment by kamilkisiela** [comment]
@manuelfink 

> I think angular2-apollo is not automatically unsubscribing from queries on ngOnDestroy (React componentWillUnmount), there as react-apollo is doing this. I guess unsubscribing is an important feature to avoid fetching data which is not needed any more.

You're right, there's no unsubscribe on ngOnDestroy. Here's why. Since `Angular2Apollo` is a service it cannot interact with your component's lifecycle automatically. In `react-apollo` they manipulate the component to assign values to properties on each Apollo's QueryObservable emission. We did the same in the decorator, but we deprecated it some time ago. BTW, this decorator will be back very soon :)
About the service. The `Angular2Apollo` service works the same as `Http` from `@angular/http`.

I exposed instance of `ApolloClient` in #220. It will be available in `0.9.0-rc.1`, today!
